sympy/matrices/matrices.py

```diff
@@ -3433,6 +3433,93 @@ def _eval_matrix_exp_jblock(self):
         from .sparsetools import banded
         return self.__class__(banded(size, bands))
 
+
+
+    def analytic_matrix(self, f, parameter = False):
```

Comment:
I think that this should be better named like `matrix_function`

Reply:
We have `M.applyfunc`, maybe we could call this `M.analytic_func`.